@here/harp-geometry/lib/OrientedBox3.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { OrientedBox3Like } from "@here/harp-geoutils";
+import { Frustum, Matrix4, Plane, Vector3 } from "three";
+
+export class OrientedBox3 implements OrientedBox3Like {
+    /**
+     * The position of the center of this `OrientedBox3`.
+     */
+    readonly position = new Vector3();
+
+    /**
+     * The x-axis of this `OrientedBox3`.
+     */
+    readonly xAxis = new Vector3(1, 0, 0);
+
+    /**
+     * The y-axis of this `OrientedBox3`.
+     */
+    readonly yAxis = new Vector3(0, 1, 0);
+
+    /**
+     * The z-axis of this `OrientedBox3`.
+     */
+    readonly zAxis = new Vector3(0, 0, 1);
+
+    /**
+     * The extents of this `OrientedBox3`.
+     */
+    readonly extents = new Vector3();
+
+    /**
+     * Creates a new `OrientedBox3`.
+     */
+    constructor();
+
+    /**
+     * Creates a new `OrientedBox3` with the given position, orientation and extents.
+     *
+     * @param position The position of the center of the `OrientedBox3`.
+     * @param rotationMatrix The rotation of the `OrientedBox3`.
+     * @param extents The extents of the `OrientedBox3`.
+     */
+    constructor(position: Vector3, rotationMatrix: Matrix4, extents: Vector3);
+
+    /**
+     * Creates a new `OrientedBox3`.
+     *
+     * @hideconstructor
+     */
+    constructor(position?: Vector3, rotationMatrix?: Matrix4, extents?: Vector3) {
```

Comment:
Minor. My intuitive priority would be `position, extents and rotation` as for many cases last one would be not used at all.

Reply:
yeah, that is definitely a possible way of defining the values, but I would prefer if we keep the order consistent with the standard way of composing/decomposing transformations, e.g.
https://threejs.org/docs/#api/en/math/Matrix4.compose
https://threejs.org/docs/#api/en/math/Matrix4.decompose

we could as well use the quaternion, instead of a Matrix, but I decided to go for the matrix4 because what we really need are the basis.

Reply:
especially for the globe, the rotation matrix would always need to be defined.